### PR TITLE
CMS Export Scalability Test, 12k Nodes -- 005

### PR DESCRIPTION
**Pull request by timwright12**
WIP

### CODE REVIEW

File: src/site/filters/liquid.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ function getPath(obj) {
 module.exports = function registerFilters() {
   const { cmsFeatureFlags } = global;
 
+  // Set timeout option to something higher (20mins)
+  // eslint-disable-next-line no-new
+  new liquid.Context({ timeout: 1200000 });
```
Comment (mpelzsherman) [inline src/site/filters/liquid.js:17]: Curious to know why this was needed.

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 2]
```diff
@@ -98,11 +99,13 @@ function pipeDrupalPagesIntoMetalsmith(contentData, files) {
...
-          pageCompiled.fieldOffice.entity.reverseFieldRegionPageNode.entities,
-        );
-        break;
+      // case 'health_services_listing':
+      //   if (pageCompiled.fieldOffice !== null) {
+      //     pageCompiled.clinicalHealthServices = sortServices(
+      //       pageCompiled.fieldOffice.entity.reverseFieldRegionPageNode.entities,
+      //     );
+      //   }
+      //   break;
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/metalsmith-drupal.js:108]: Why was this removed? Without sorting, do the services still appear in alphabetical order on the page (e.g. /pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive)?

File: src/site/filters/liquid.js (hunk 1 above)
Comment (jhonnyoddball) [inline src/site/filters/liquid.js:17]: Do you think this will be beneficial to master?

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 3]
```diff
@@ -59,13 +59,14 @@ function pipeDrupalPagesIntoMetalsmith(contentData, files) {
     const pageCompiled = compilePage(page, contentData);
     const drupalPageDir = path.join('.', drupalUrl);
     const drupalFileName = path.join(drupalPageDir, 'index.html');
+    const layoutName = entityBundle || 'page';
 
     switch (page.entityBundle) {
       case 'health_care_local_facility':
-        addGetUpdatesFields(pageCompiled, pages);
+        // addGetUpdatesFields(pageCompiled, pages);
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/metalsmith-drupal.js:66]: According to the comments on addGetUpdatesFields, this is needed to add social links to these pages.

Are we sure this isn't needed?

File: src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js [hunk 4]
```diff
@@ -11,16 +11,16 @@ module.exports = {
...
+    // field_facility_service_hours: {
+    //   type: 'array',
+    //   items: {
+    //     type: 'object',
+    //     properties: {
+    //       value: { type: 'array' },
+    //     },
+    //     maxItems: 1,
+    //   },
+    // },
```
Comment (mpelzsherman) [inline src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js:23]: Why was this removed?

File: src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js [hunk 5]
```diff
@@ -6,15 +6,23 @@ const getFieldFacilityLocationObject = ({
...
+  return typeof title === 'object'
     ? {
         entity: {
           title: getDrupalValue(title),
           fieldFacilityLocation: {
             entity: {
-              title: getDrupalValue(field_facility_location[0].title),
-              entityUrl: field_facility_location[0].entityUrl,
+              title: getDrupalValue(fieldFacilityTitle),
+              entityUrl: fieldFacilityUrl,
```
Comment (mpelzsherman) [inline src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js:25]: Curous about these as well.

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 6]
```diff
@@ -59,13 +59,14 @@ function pipeDrupalPagesIntoMetalsmith(contentData, files) {
     const pageCompiled = compilePage(page, contentData);
     const drupalPageDir = path.join('.', drupalUrl);
     const drupalFileName = path.join(drupalPageDir, 'index.html');
+    const layoutName = entityBundle || 'page';
```
Comment (jhonnyoddball) [inline src/site/stages/build/drupal/metalsmith-drupal.js:62]: Is this an improvement or just for testing? If so, why?

File: src/site/stages/build/process-cms-exports/schemas/output/node-regional_health_care_service_des.js [hunk 7]
```diff
@@ -12,10 +12,11 @@ module.exports = {
       type: 'array',
       items: {
         entity: {
-          title: { type: 'string' },
+          title: { type: ['string', 'TBD'] },
```
Comment (mpelzsherman) [inline src/site/stages/build/process-cms-exports/schemas/output/node-regional_health_care_service_des.js:15]: I don’t think TBDis a valid output type

File: src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js (hunk 4 above)
Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js:23]: Is it to bypass an error?

File: src/site/stages/build/process-cms-exports/schemas/output/node-health_services_listing.js [hunk 8]
```diff
@@ -11,7 +11,7 @@ module.exports = {
     promote: { type: 'boolean' },
     sticky: { type: 'boolean' },
     defaultLangcode: { type: 'boolean' },
-    revisionTranslationAffected: { type: 'boolean' },
+    revisionTranslationAffected: { type: ['boolean', 'null'] },
```
Comment (jhonnyoddball) [inline src/site/stages/build/process-cms-exports/schemas/output/node-health_services_listing.js:14]: This is fine

File: src/site/filters/liquid.js (hunk 1 above)
Comment (timwright12) [reply src/site/filters/liquid.js:17]: @mpelzsherman as the node count increases the parsing time for templates also goes up. The TinyLiquid plugin we're using has a 2min timeout, so I increased it

Comment (timwright12) [reply src/site/filters/liquid.js:17]: @jhonnyoddball nothing in the current state will, but it's more so looking down the line and noting the errors we saw, and planning for them now

File: src/site/stages/build/drupal/metalsmith-drupal.js (hunk 6 above)
Comment (timwright12) [reply src/site/stages/build/drupal/metalsmith-drupal.js:62]: @jhonnyoddball we were getting build failures when the entityBundle was returning null, setting the default template to page fixed that

File: src/site/stages/build/drupal/metalsmith-drupal.js (hunk 3 above)
Comment (timwright12) [reply src/site/stages/build/drupal/metalsmith-drupal.js:66]: This was just failing our tests, wouldn't be something merged

File: src/site/stages/build/drupal/metalsmith-drupal.js (hunk 2 above)
Comment (timwright12) [reply src/site/stages/build/drupal/metalsmith-drupal.js:108]: just failing tests, we wouldn't merge this part

File: src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js (hunk 4 above)
Comment (timwright12) [reply src/site/stages/build/process-cms-exports/schemas/input/paragraph-service_location.js:23]: failing builds with the tests data, wouldn't merge this either

File: src/site/stages/build/process-cms-exports/schemas/output/node-regional_health_care_service_des.js (hunk 7 above)
Comment (timwright12) [reply src/site/stages/build/process-cms-exports/schemas/output/node-regional_health_care_service_des.js:15]: you can ignore this, we were testing something

File: src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js (hunk 5 above)
Comment (timwright12) [reply src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js:25]: This was @k80bowman's update, let me get her

Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js:25]: Curious too.

Looks good but the more fields needed in a future will make it harder to understand. Needed to be done in the `else` statement below as well for consistency of both ends

File: src/site/filters/liquid.js (hunk 1 above)
Comment (mpelzsherman) [reply src/site/filters/liquid.js:17]: OK. I don't know enough about how we're using TinyLiquid to understand why we are hitting this limit. Is it timing out while rendering a single page, or does the timeout need to cover the total time to render all pages?

File: src/site/stages/build/drupal/graphql/navigation-fragments/menuLinks.nav.graphql.js [hunk 9]
```diff
@@ -7,7 +7,7 @@
 const headerQuery = require('./header.nav.graphql');
 
 module.exports = `
-  menuLinkContentQuery(limit: 5000, filter: {conditions: [{field: "enabled", value: "1"}]}) {
+  menuLinkContentQuery(limit: 15000, filter: {conditions: [{field: "enabled", value: "1"}]}) {
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/graphql/navigation-fragments/menuLinks.nav.graphql.js:10]: This is probably OK to merge to master.

File: src/site/stages/build/drupal/graphql/GetAllPages.graphql.js [hunk 10]
```diff
@@ -97,7 +97,7 @@ const buildQuery = ({ useTomeSync }) => {
   const nodeQuery = useTomeSync
     ? ''
     : `
-    nodeQuery(limit: 5000, filter: {
+    nodeQuery(limit: 15000, filter: {
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/graphql/GetAllPages.graphql.js:100]: Probably OK to merge to master, but we may run into memory issues.

File: src/site/stages/build/process-cms-exports/schemas/output/node-health_services_listing.js [hunk 11]
```diff
@@ -23,7 +23,7 @@ module.exports = {
     fieldIntroText: { type: 'string' },
     fieldMetaTitle: { type: 'string' },
     fieldOffice: {
-      type: 'object',
+      type: ['object', 'null'],
```
Comment (mpelzsherman) [inline src/site/stages/build/process-cms-exports/schemas/output/node-health_services_listing.js:26]: I think this is OK

File: src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js (hunk 5 above)
Comment (k80bowman) [reply src/site/stages/build/process-cms-exports/transformers/node-regional_health_care_service_des.js:25]: There were instances in some of the test data where there was no facility title or entityUrl in that first array object of field_facility_location. So, in order to get the build to pass, I added these checks that would look for the keys in that first array object and, if it couldn't find them, set the variable to an empty string.

I don't know if this is going to be a thing when this data is published to prod, so any insights there or modifications needed are welcome. I didn't dig too much into making these changes consistent or ready for long-term use. They were purely intended to get the test builds to pass.

File: src/site/filters/liquid.js (hunk 1 above)
Comment (timwright12) [reply src/site/filters/liquid.js:17]: It only happened with the node increase to 12k, so very likely the time to render all pages unless there's a single gigantic page buried in there somewhere

File: src/site/stages/build/drupal/graphql/GetAllPages.graphql.js (hunk 10 above)
Comment (timwright12) [reply src/site/stages/build/drupal/graphql/GetAllPages.graphql.js:100]: I'll hold this one back

File: src/site/stages/build/drupal/graphql/navigation-fragments/menuLinks.nav.graphql.js (hunk 9 above)
Comment (timwright12) [reply src/site/stages/build/drupal/graphql/navigation-fragments/menuLinks.nav.graphql.js:10]: same, will hold this back
